### PR TITLE
AdminController delete user (#1)

### PR DESCRIPTION
**Pull request by avigoen**
Delete user from Quora Application

### CODE REVIEW

File: quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    private UserDao userDao;
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public Integer deleteUser(final String userUuid, final String authToken) throws AuthorizationFailedException, UserNotFoundException {
+        UserAuthTokenEntity userAuthTokenEntity = userDao.getUserAuthToken(authToken);
+        if (userAuthTokenEntity == null) {
+            throw new AuthorizationFailedException("ATHR-001", "User has not signed in");
+        }
+
+        if (userAuthTokenEntity.getLogoutAt() != null) {
```
Comment (manjunath84) [inline quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java:26]: I think you need to check if the JWT token is expired or if logout time is before the current time. I have written below logic while implementing similar functionality.

 final ZonedDateTime now = ZonedDateTime.now();
        //User is signed out if either JWT token is expired or user has logged out
        if(userAuthTokenEntity.getExpiresAt().isBefore(now) ||
                (userAuthTokenEntity.getLogoutAt()!=null && userAuthTokenEntity.getLogoutAt().isBefore(now))){

File: quora-api/src/main/java/com/upgrad/quora/api/controller/UserAdminController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+@RestController
+@RequestMapping("/")
+public class UserAdminController {
+
+    @Autowired
+    private UserAdminBusinessService userAdminBusinessService;
+
+    @RequestMapping(value = "/admin/user/{userId}", method = RequestMethod.DELETE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<UserDeleteResponse> deleteUser(@PathVariable("userId") final String userUuid, final String authorization) throws AuthorizationFailedException, UserNotFoundException {
```
Comment (manjunath84) [inline quora-api/src/main/java/com/upgrad/quora/api/controller/UserAdminController.java]: Please add method level comments here.

File: quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            throw new AuthorizationFailedException("ATHR-002", "User is signed out");
+        }
+
+        UserEntity userEntity = userAuthTokenEntity.getUser();
+        String userRole = userEntity.getRole();
```
Comment (manjunath84) [inline quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java:31]: You can add some comments to explain the functionality here.

File: quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class UserAdminBusinessService {
+
+    @Autowired
+    private UserDao userDao;
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public Integer deleteUser(final String userUuid, final String authToken) throws AuthorizationFailedException, UserNotFoundException {
```
[the remark this answers is not in the document]
Comment (avigoen) [reply quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java:20]: Comments added

File: quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java (hunk 1 above)
Comment (avigoen) [reply quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java:26]: Comments added

File: quora-api/src/main/java/com/upgrad/quora/api/controller/UserAdminController.java (hunk 2 above)
Comment (avigoen) [reply quora-api/src/main/java/com/upgrad/quora/api/controller/UserAdminController.java]: Comments added

File: quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java (hunk 3 above)
Comment (avigoen) [reply quora-service/src/main/java/com/upgrad/quora/service/business/UserAdminBusinessService.java:31]: Comments added

**Pull request merged**